test/crawlers/cheerio_crawler.js

```diff
@@ -416,18 +425,31 @@ describe('CheerioCrawler', () => {
                 await crawler.run();
 
                 expect(handlePageInvocationCount).to.be.eql(0);
-                expect(errorMessages).to.have.lengthOf(4);
-                errorMessages.forEach(msg => expect(msg).to.include('aborted due to abortFunction'));
+                expect(errorMessages).to.have.lengthOf(8);
+                errorMessages.forEach(msg => expect(msg).to.include('Internal Server Error'));
             });
 
             it('when statusCode >= 500 and application/json is received', async () => {
+                const save = sinon.stub(utilsRequest, 'requestAsBrowser');
```

Comment:
Why not use the Express server too for this test?

Reply:
I am not sure why, but I am having troubles with express to send application/json headers. it overrides them to the text/html or text/plain at the end. I was debugging it and found no reason 😞